arbor/backends/multicore/shared_state.cpp

```diff
@@ -168,27 +169,34 @@ void shared_state::ions_init_concentration() {
 }
 
 void shared_state::update_time_to(fvm_value_type dt_step, fvm_value_type tmax) {
+    using simd::simd_cast;
+    using simd::indirect;
+    using simd::add;
+    using simd::min;
     for (fvm_size_type i = 0; i<n_intdom; i+=simd_width) {
-        simd_value_type t(time.data()+i);
-        t = min(t+dt_step, simd_value_type(tmax));
-        t.copy_to(time_to.data()+i);
+        simd_value_type t = simd_cast<simd_value_type>(indirect(time.data()+i, simd_width));
```

Comment:
I prefer the first method, I'll add an `assign` function to the API. This will require quite a few changes to integrate into the cprinter correctly. Maybe we can add an issue for that?